src/main/java/seedu/address/ui/PersonInfoPanel.java

```diff
@@ -152,17 +177,43 @@ public void loadPhoto(ReadOnlyPerson person) {
 
     }
 
-
-
     //@@author zacharytang
     @Subscribe
     private void handlePersonSelectedEvent(PersonSelectedEvent event) {
         logger.info(LogsCenter.getEventHandlingLogMessage(event));
-        //this.person = person;
-        //initTags(person);
         loadPerson(event.person);
     }
 
+    //@@author nbriannl
+    @Subscribe
+    private void handlePersonHasBeenModifiedEvent(PersonHasBeenModifiedEvent event) {
+        logger.info(LogsCenter.getEventHandlingLogMessage(event));
+        if (currentlyViewedPerson != null && currentlyViewedPerson.equals(event.oldPerson)) {
+            loadPerson(event.newPerson);
+        }
+    }
+
+    //@@author nbriannl
+    @Subscribe
+    private void handlePersonHasBeenDeletedEvent(PersonHasBeenDeletedEvent event) {
+        logger.info(LogsCenter.getEventHandlingLogMessage(event));
+        if (currentlyViewedPerson != null && currentlyViewedPerson.equals(event.deletedPerson)) {
+            clearBind();
+            loadDefaultPerson();
+        }
+    }
+
+    //@@author nbriannl
+    @Subscribe
+    private void handleAddressBookChangedEvent(AddressBookChangedEvent event) {
+        logger.info(LogsCenter.getEventHandlingLogMessage(event));
+        if (event.data.getPersonList().size() == 0 && event.data.getTagList().size() == 0) {
+            clearBind();
+            loadDefaultPerson();
+        }
+    }
+
```

Comment:
The 3 events that trigger this, and the app is look out for is
- Person has been modified via Edit or Delete Tag
-- Check if the modified person is the currentlyViewedPerson
-- Then load the new person.
- Person has been deleted
-- Check if the deleted person is the currentlyViewedPerson
-- Then load a default blank person
- AddressBook has been changed
-- Check if the address book is empty
-- Then load a default person